tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedReadHandleImpl.java

```diff
@@ -55,6 +57,11 @@ public class BlobStoreBackedReadHandleImpl implements ReadHandle {
     private final BackedInputStream inputStream;
     private final DataInputStream dataStream;
     private final ExecutorService executor;
+    // this Cache is accessed only by one thread
+    private final Cache<Long, Long> entryOffsets = CacheBuilder
+            .newBuilder()
+            .expireAfterAccess(10, TimeUnit.MINUTES)
```

Comment:
10 minutes may be too short, 30 minutes to 1 hour?  This cache can be shared with other catch-up readers.

Reply:
I have changed it to 30 minutes and made it configurable via system property
I don't think we have to document the system property, I made it configurable only in case of problems